### DISCUSSION TITLE
Wiki.js on headless server install can’t finish install because of browser requirement

### DISCUSSION THREAD

**Discussion by mapes62** [requarks/wiki#4691] [Help / Questions] [answered]
I am probably making my life more complicated than it needs to be; but, I am trying to install wiki.js on a headless server. 

I have everything configured as I assume correctly the issue is during the installation it asks me to navigate my browser to http://your-server-ip:port. 

Seeing as I don’t have access to a browser on my localhost in this setup is there a way to complete the install from my remote machine that has access to a browser? I tried using Nginx to serve the website but that isn’t working either. Please see below for copies of my configurations for both wiki.js and nginx.

I am not 100% certain how to include the "path" for the nginx server to find the appropriate webpage to host from wiki.js or make that connection which may be causing my issues.

Appreciate any help anyone can provide. 

# NGINX Configuration:

        user nginx;
        worker_processes auto;
        error_log /var/log/nginx/error.log;
        pid /run/nginx.pid;

        include /usr/share/nginx/modules/*.conf;

        events {
            worker_connections 1024;
        }

        http {
            log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
                              '$status $body_bytes_sent "$http_referer" '
                             '"$http_user_agent" "$http_x_forwarded_for"';

           access_log /var/log/nginx/access.log main;

           sendfile                        on;
           tcp_nopush                  on;
           tcp_nodelay                 on;
           keepalive_timeout        65;
           types_hash_max_size 4096;

           include                         /etc/nginx/mime.types;
           default_type                 application/octet-stream;

           include  /etc/nginx/conf.d/*.conf;

           server  {
                 listen               80;
                 listen               [::]:80;
                 server_name  _;
                 root                 /usr/share/nginx/html;

                 include /etc/nginx/default.d/*.conf;

                 location / {
                       proxy_set_header Host $http_host;
                       proxy_set_header X-Real_IP $remote_addr;
                       proxy_pass http://127.0.0.1:3000;
                       proxy_http_version 1.1;
                       proxy_set_header Upgrade $http_upgrade;
                       proxy_set_header Connection "upgrade";
                       proxy_next_upstream error timeout http_502 http_503 http_504;
               }
 
                error_page 404 /404.html;
                      location = /40X.html {
               }

               error_page 500 502 503 504 /50x.html;
                    location = /50x.html {
               }
            }

            server {
                listen       443 ssl http2;
                listen       [::]:443 ssl http2;
                server_name  _;
                root         /usr/share/nginx/html;

                ssl_certificate "/etc/pki/server.crt";
                ssl_certificate_key "/etc/pki/server.key";

                ssl_session_cache shared:SSL:50m;
                ssl_session_timeout  1d;
                ssl_ciphers PROFILE=SYSTEM;

                ssl_protocols TLSv1.1 TLSv1.2 TLSv1.3;
                ssl_prefer_server_ciphers on;
                ssl_session_tickets off;

                include /etc/nginx/default.d/*.conf;

                location / {
                    proxy_set_header Host $http_host;
                    proxy_set_header X-Real_IP $remote_addr;
                    proxy_pass http://127.0.0.1:3000;
                    proxy_http_version 1.1;
                    proxy_set_header Upgrade $http_upgrade;
                    proxy_set_header Connection "upgrade";
                    proxy_next_upstream error timeout http_502 http_503 http_504;
                }

        	error_page 404 /404.html;
                    location = /40x.html {
                }

                error_page 500 502 503 504 /50x.html;
                    location = /50x.html {
                }
           }

        }

# Wiki.js Configuration:

        # ---------------------------------------------------------------------
        # Port the server should listen to  
        # ---------------------------------------------------------------------

        port: 3000

        # ---------------------------------------------------------------------
        # Database
        # ---------------------------------------------------------------------
        db:
           type: postgres

         # PostgreSQL / MySQL / MariaDB / MS SQL Server only:
          host: localhost
          port: 5432
          user: wikijs
          pass: secret
         db: wikijsdb
         ssl: false

        sslOptions:
            auto: true
         schema: public

        # ---------------------------------------------------------------------
        # IP address the server should listen to
        # ---------------------------------------------------------------------
        bindIP: 127.0.0.1
        # ---------------------------------------------------------------------
        # Log Level
        # ---------------------------------------------------------------------
        logLevel: info
        # ---------------------------------------------------------------------
        # Offline Mode
        # ---------------------------------------------------------------------
        offline: false
        # ---------------------------------------------------------------------
        # High-Availability
        # ---------------------------------------------------------------------
        ha: false
        # ---------------------------------------------------------------------
        # Data Path
        # ---------------------------------------------------------------------
        dataPath: ./data

**Comment by ieaster1** [comment, marked as answer]
You don't _need_ nginx serving the web app unless this is a requirement for you.

You will want to change your wiki config:
```
bindIP: 127.0.0.1
```
To: 
```
bindIP: 0.0.0.0
```

Then restart the node server.

This will allow external access to the web application. You will be able to access it at http://<server_ip_addr>:3000/

If you're still having issues, you'll want to check and make sure you see that node has bound to 0.0.0.0:3000.  You'd also want to check your firewall if you're running one.

You can use `netstat` and grep for port  3000:
```
/wiki $ netstat -tulpn | grep 3000
tcp        0      0 0.0.0.0:3000            0.0.0.0:*               LISTEN      1/node
```

**Reply by mapes62** [reply]
Thank you very much! this has been driving me nuts for a few days of troubleshooting many different things and has worked.  Appreciate the quick reply to this issue.
